src/index.ts

```diff
@@ -11,19 +11,19 @@ import {
   WebhookError,
   HandlerFunction,
 } from "./types";
-import { EventNames } from "./generated/event-names";
 import { IncomingMessage, ServerResponse } from "http";
+import { EventNames } from "./generated/event-names";
 
-class Webhooks<T extends WebhookEvent = WebhookEvent> {
+class Webhooks<T extends WebhookEvent = WebhookEvent, U = {}> {
```

Comment:
What exactly is the U here?

Reply:
U represents the additional properties that the transform function returns apart from the event payload object. Used here https://github.com/ankeetmaini/webhooks.js/blob/4e8d65b891f4adaa1426eb505a92b321c307bb2f/test/typescript-validate.ts#L124

Reply:
I think before the type was inferred automatically from the return type of the transform method, is that still the case?

Reply:
I'm not sure how that can be achieved. I can try and move the `Option<T>` param up so that it can be passed down to `on` handler and then try and extract the return type of `Option<T>['transform']` but since the `transform` function would return an intersection type like `WebhookEvent<any> & {some: 'other', type: 'attributes'}` I can't use a conditional value to extract the second part. Mostly because intersection types are not distributed like union ones.

Also if I take the entire value, then intellisense goes out the window as `WebhookEvent<any>` takes precedence.

Reply:
This can probably be solved by changing the Webhook<T> signature, right now T is the actual payload, if we're to map that as the event  name, then we can also theoretically get the value returned automatically. But this would be a big change like I was trying to attempt in #258 and I was all over the place 😖

Reply:
Okay, let's leave it as is, we can do a follow up PR if we come up with something clever :) 

Can you please add a comment explaining the `U` type parameter?

Reply:
I've added the comment! I'll try and see if I can improve the types further in a follow up PR :D